autoload/over/command_line/search.vim

```diff
@@ -16,29 +17,60 @@ function! s:search_hl_off()
 		call matchdelete(s:search_id)
 		unlet s:search_id
 	endif
+	if exists("s:current_search_id") && s:current_search_id != -1
+		call matchdelete(s:current_search_id)
+		unlet s:current_search_id
+	endif
+	if exists("s:cursor_search_id") && s:cursor_search_id != -1
+		call matchdelete(s:cursor_search_id)
+		unlet s:cursor_search_id
+	endif
 endfunction
 
 
 function! s:search_hl_on(pattern)
 	call s:search_hl_off()
 	silent! let s:search_id = matchadd("IncSearch", a:pattern)
+	silent! let s:current_search_id = matchadd(
+		\ "Search",
+		\ '\%#' . a:pattern
+	\ )
+	silent! let s:cursor_search_id = matchadd("Cursor", '\%#')
 endfunction
 
 
 function! s:main()
 	call s:search_hl_off()
 	let line = over#command_line#backward()
+
+	let visual = 0
+	if line[0:4] == "'<,'>"
+		let visual = 1
+		let line = line[5:]
+	endif
+
 	if line =~ '^/.\+'
 \	|| line =~ '^?.\+'
-		let pattern = matchstr(line, '^\(/\|?\)\zs.\+')
+
+		let s:pattern = matchstr(line, '^\(/\|?\)\zs.\+')
+
+		if visual
+			let s:pattern = over#command_line#bind_pattern_by_visual(
+				\ s:pattern
+			\ )
+		endif
+
+		nohlsearch
+
 		if g:over#command_line#search#enable_incsearch
-			call s:search_hl_on((&ignorecase ? '\c' : "") . pattern)
+			call s:search_hl_on((&ignorecase ? '\c' : "") . s:pattern)
 		endif
 		if g:over#command_line#search#enable_move_cursor
+			call setpos(".", s:old_pos)
 			if line =~ '^/.\+'
-				silent! call search(pattern, "c")
+				silent! call searchpos(s:pattern, "")
 			else
-				silent! call search(pattern, "cb")
+				silent! call searchpos(s:pattern, "b")
```

Comment:
Because it's expected behaviour when searching. I expect cursor to be moved on the first search result, not just highlight the matches. Like usual searching through `/` works.